packages/validator/src/services/doppelgangerService.ts

```diff
@@ -54,16 +58,41 @@ export class DoppelgangerService {
     this.logger.info("Doppelganger protection enabled", {detectionEpochs: DEFAULT_REMAINING_DETECTION_EPOCHS});
   }
 
-  registerValidator(pubkeyHex: PubkeyHex): void {
+  async registerValidator(pubkeyHex: PubkeyHex): Promise<void> {
     const {currentEpoch} = this.clock;
     // Disable doppelganger protection when the validator was initialized before genesis.
     // There's no activity before genesis, so doppelganger is pointless.
-    const remainingEpochs = currentEpoch <= 0 ? 0 : DEFAULT_REMAINING_DETECTION_EPOCHS;
+    let remainingEpochs = currentEpoch <= 0 ? REMAINING_EPOCHS_IF_SKIPPED : DEFAULT_REMAINING_DETECTION_EPOCHS;
     const nextEpochToCheck = currentEpoch + 1;
 
     // Log here to alert that validation won't be active until remainingEpochs == 0
     if (remainingEpochs > 0) {
-      this.logger.info("Registered validator for doppelganger", {remainingEpochs, nextEpochToCheck, pubkeyHex});
+      const previousEpoch = currentEpoch - 1;
+      const attestedInPreviousEpoch = await this.slashingProtection.hasAttestedInEpoch(
+        fromHexString(pubkeyHex),
+        previousEpoch
+      );
+
+      if (attestedInPreviousEpoch) {
```

Comment:
consider committing this explanation into the code

Reply:
I mostly added this here to get a second opinion on this if my reasoning is sound. But is might be a good idea to add this to the issue description for better visibility instead of here.

The issue is already referenced in the code
https://github.com/ChainSafe/lodestar/blob/eaf74cda4c43a83ebc184d096e08a599420754b9/packages/validator/src/services/doppelgangerService.ts#L76-L78

Will refine the issue description a bit